crawler/urlJobProcessor.js

```diff
@@ -1,36 +1,124 @@
+    hasManifest: false,
+  };
+
+  // TODO: Pass user agent in headers
+  const fetchParams = {
+    headers: '',
+  };
+
+  return fetch(httpUrl.href, fetchParams).catch(() => {
+    // Ignore errors trying to fetch HTTP
+    return null;
+  }).then((httpResponse) => {
+    if (httpResponse) {
+      // Check the HTTP response for a W3C App Manifest
+      return httpResponse.text().then((htmlText) => {
+        const $ = cheerio.load(htmlText);
+        $('link[rel="manifest"]').each(() => {
+          ret.hasManifest = true;
+          return false;
+        });
+      }).catch((err) => {
+        // Log and ignore issues when parsing HTTP response
+        console.error(`Unexpected error parsing HTTP response: ${err}`);
+      }).then(() => {
+        // Check whether we were redirected to HTTPS
+        const responseUrl = urlModule.parse(httpResponse.url);
+        if (responseUrl.protocol === 'https:') {
+          ret.hasHTTPSRedirect = true;
+          return httpResponse;
+        }
+
+        return null;
+      });
+    }
+  }).then((httpsResponse) => {
+    // If we already fetched over HTTPS thanks to an HTTP redirect
+    // just keep the previous response. If not, we have to fetch over
+    // HTTPS now.
+    if (!httpsResponse) {
+      return fetch(httpsUrl.href, fetchParams);
+    }
+    return httpsResponse;
+  }).then((httpsResponse) => {
+    ret.hasHTTPS = true;
+    return httpsResponse.text();
+  }).then((htmlText) => {
+    const $ = cheerio.load(htmlText);
+
+    // Check the HTTPS response for a W3C App Manifest
+    $('link[rel="manifest"]').each(() => {
+      ret.hasManifest = true;
+      return false;
+    });
+
+    const scripts = [];
+    const fetchPromises = [];
+    $('script').each((index, elem) => {
+      const src = $(elem).attr('src');
+      if (!src) {
+        scripts.push($(elem).text());
+      } else {
+        const resolvedSrc = urlModule.resolve(httpsUrl.href, src);
+        fetchPromises.push(fetch(resolvedSrc).catch(() => {
+          // Ignore errors trying to fetch individual scripts
+        }).then((scriptResponse) => {
+          return scriptResponse.text();
+        }).then((jsText) => {
+          return scripts.push(jsText);
+        }));
+      }
     });
-  }).then(() => {
-    // TODO: Fetch dependent resources (e.g. js, serviceWorker)
-    // and store them in the fields of `ret`
-    return Promise.resolve();
-  }).then(() => {
-    return ret;
-  });
-}
 
-function processUrlJob(urlJob) {
-  console.log(`processUrlJob - ${urlJob}`);
+    return Promise.all(fetchPromises).then(() => scripts);
+  }).then((scripts) => {
+    const swStr = '.serviceWorker';
+    const regStr = '.register';
 
-  return fetchAllResources(urlJob).then((pageResources) => {
-    const promises = [];
-    probes.forEach((probeFunc) => {
-      promises.push(probeFunc.apply(null, pageResources));
+    let swMatched = false;
+    let regMatched = false;
+
+    scripts.forEach((script) => {
```

Comment:
I'd expect both occurrences in the same file. Devs will not assign serviceWorker in one module and register in another. This would shorten this check into one `if`.